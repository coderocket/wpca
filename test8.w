proc Max[x,y,z:int]
	{ x = X and y = Y }
; if x > y -> z := x [] x=y -> z := (x+y) div 2 [] x < y -> z := y fi
	{ x = X and y = Y and z = max[x,y] }

