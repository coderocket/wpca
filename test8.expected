open misc
open util/integer

pred true { no none }
pred false { some none }

one sig State {
 x,y,z : Int
}

one sig Const {
 X : Int,
Y : Int
}

pred obligation {
(State.x = Const.X and State.y = Const.Y) => {
(State.x > State.y or (State.x = State.y or State.x < State.y))
(State.x > State.y => State.x = Const.X and State.y = Const.Y and State.x = (State.y).((State.x).(max)))
(State.x = State.y => State.x = Const.X and State.y = Const.Y and State.x.add[State.y].div[2] = (State.y).((State.x).(max)))
(State.x < State.y => State.x = Const.X and State.y = Const.Y and State.y = (State.y).((State.x).(max)))
}
}

check { obligation }
